dbt/adapters/fabricspark/livysession.py

```diff
@@ -472,6 +476,10 @@ def connect(credentials: SparkCredentials) -> LivyConnection:
             __class__.livy_global_session = LivySession(credentials)
             __class__.livy_global_session.create_session(data)
             __class__.livy_global_session.is_new_session_required = False
+            # create shortcuts, if there are any
+            if credentials.shortcuts_json_path:
+                shortcut_client = ShortcutClient(accessToken.token, credentials.workspaceid, credentials.lakehouseid)
+                shortcut_client.create_shortcuts(credentials.shortcuts_json_path, retry=credentials.retry_all)

```

Comment:
I suggest a get shortcut before creating a shortcut to check if shortcut exists. Also, if all values in a shortcut match with json config, do not create a shortcut.
conflict policy should be added as a query string parameter to create shortcut api call.. default shortcut conflict policy should be abort.

https://learn.microsoft.com/en-us/rest/api/fabric/core/onelake-shortcuts/create-shortcut?tabs=HTTP#shortcutconflictpolicy

Reply:
i'll add the get check.

for the conflict policy, i believe it defaults to abort without including the param -- do we want to include abort specifically or are you suggesting we allow the user to specify the conflict policy?

Reply:
If default is abort, then we should leave it as is. We will not use "generate unique shortcut name" option because that can lead to duplicate data sources for the same shortcut source